Dockerfile

```diff
@@ -1,24 +1,28 @@
-FROM node:14.18.0-alpine as build
+FROM node:16.15.1-alpine as build
 
 ARG COMMIT_SHA=<not-specified>
-ENV NODE_ENV=production
 
 WORKDIR /build-dir
 
 COPY package.json .
 COPY package-lock.json .
 
-RUN npm ci
+# install also DEV dependencies to enable Typescript compilation
+RUN npm ci --production=false
 
 COPY . .
 
+# compile Typescript to Javascript
 RUN npm run build
 
+# remove files not needed in the final stage
+RUN rm -rf node_modules src tsconfig.json .swcrc
+
 RUN echo "mia_template_service_name_placeholder: $COMMIT_SHA" >> ./commit.sha
 
 ########################################################################################################################
 
-FROM node:14.18.0-alpine
+FROM node:16.15.1-alpine
```

Comment:
this version of node is not compatible with how we define k8s cluster users; since npm introduced a constraint on the user; we have to set the variable:

```
ENV NPM_CONFIG_CACHE="/tmp"
```